xswap/xswap.py

```diff
@@ -42,7 +42,13 @@ def permute_edge_list(edge_list: List[Tuple[int, int]], allow_self_loops: bool =
     """
     max_source = max([i[0] for i in edge_list])
     max_target = max([i[1] for i in edge_list])
+    max_source, max_target = sorted([max_source, max_target])
```

Comment:
What is this for?

Reply:
There is an issue whereby if `max_source` > `max_target`, then the amount of memory allocated for the bitset is insufficient. This stems from the fact that the Cantor pairing function depends on the order of inputs.